File: solutions/typescript/armstrong-numbers/1/armstrong-numbers.ts
export function isArmstrongNumber(number: number | BigInt): boolean {
  if (typeof number === 'number') {
    if (number < 0) {
      return false
    }
    const numStr = number.toString()
    const numDigits = numStr.length
    return number === numStr.split('').reduce((acc, digit) => acc + Number(digit) ** numDigits, 0)
  } else {
    const numStr = number.toString()
    const numDigits = BigInt(numStr.length)
    return number === numStr.split('').reduce((acc, digit) => acc + BigInt(digit) ** numDigits, 0n)
  }
}

File: solutions/python/armstrong-numbers/1/armstrong_numbers.py
def sum_of_powers(number, power):
    sum = 0
    for digit in str(number):
        sum += pow(int(digit), power)
    return sum


def is_armstrong_number(number):
    number_of_digits = len(str(number))
    return number == sum_of_powers(number, number_of_digits)
